sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala

```diff
@@ -346,36 +319,50 @@ trait ProverInterpreter extends Interpreter with AttributionCore {
     t.withPolynomial(q).copy(children=newChildren)
     */
 
-
-
     case su: UnprovenSchnorr =>
-      if (su.simulated) {
-        // Step 5 (simulated leaf -- complete the simulation)
-        assert(su.challengeOpt.isDefined)
-        val (fm, sm) = DLogInteractiveProver.simulate(su.proposition, su.challengeOpt.get)
-        UncheckedSchnorr(su.proposition, Some(fm), su.challengeOpt.get, sm)
-      } else {
-        // Step 6 (real leaf -- compute the commitment a)
-        val (r, commitment) = DLogInteractiveProver.firstMessage(su.proposition)
-        su.copy(commitmentOpt = Some(commitment), randomnessOpt = Some(r))
+      //Steps 5 & 6: pull out commitment from the hints bag, otherwise, compute the commitment(if the node is real),
+      // or simulate it (if the node is simulated)
+
+      // Step 6 (real leaf -- compute the commitment a or take it from the hints bag)
```

Comment:
Step 6 comment seems to be incorrect since it handles both real and simulated. Perhaps it should be `First try pulling out commitment from hints bag if it exists`

Reply:
well, no, step 6 is for the real leafs only

Reply:
actually there are two statements referring to step 6. Maybe its better to separate them as 6-a?
// Step 6 (real leaf -- compute the commitment a or take it from the hints bag)
// Step 6 -- compute the commitment